pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -225,6 +229,28 @@ func (r *repository) Put(ctx context.Context, manifest *manifest.SignedManifest)
 	return nil
 }
 
+// createImageStreamMapping creates ISM, the operation is retried if conflict at ImageStream
+// level is returned as a result, in all other cases the actual error is returned.
+func (r *repository) createImageStreamMapping(namespace string, ism *imageapi.ImageStreamMapping, retryCount int) error {
+	if retryCount == ismRetries {
+		return fmt.Errorf("reched ismRetries limit")
+	}
+	if err := r.registryClient.ImageStreamMappings(r.namespace).Create(ism); err != nil {
+		// if the error was image stream update conflict try re-invoking ism creation
+		statusErr, ok := err.(*kerrors.StatusError)
+		if !ok {
+			return err
+		}
+
+		status := statusErr.ErrStatus
+		if status.Code != http.StatusConflict || status.Details.Kind != "imageStream" || status.Details.Name != ism.Name {
+			return err
+		}
+		return r.createImageStreamMapping(namespace, ism, retryCount+1)
```

Comment:
I find the incrementation of retry counter confusing. When a func is invoked with `retrycount == 0`, I'd expect it to fail without a retry. I'd rather follow our pattern in various controllers where the counter is decremented until it reaches 0.